tests/Unit/Core/Domain/Carrier/Command/TestAddCarrierCommand.php

```diff
@@ -0,0 +1,231 @@
+        yield [0, 0, 0, -0.1];
+        yield [-1, -2, -3, -5.5];
+    }
+
+    public function getInvalidRanges()
+    {
+        yield [[]];
+        yield [[
+            'assoc' => [
+                'from' => 1,
+                'to' => 2,
+                'prices_by_zone_id' => [
+                    3 => 1,
+                    4 => 2,
+                ],
+            ],
+        ]];
+        yield [[
+            [
+                'to' => 2,
+                'prices_by_zone_id' => [
+                    3 => 1,
+                    4 => 2,
+                ],
+            ],
+        ]];
+        yield [[
+            [
+                'from' => 2,
+                'prices_by_zone_id' => [
+                    3 => 2,
+                    5 => 1,
+                ],
+            ],
+        ]];
+        yield [[
+            [
+                'from' => 1,
+                'to' => 2,
+            ],
+        ]];
+        yield [[
+            [
+                'from' => 2,
+                'to' => 3,
+                'prices_by_zone_id' => [],
+            ],
+        ]];
+        yield [[
+            [
+                'from' => 1,
+                'to' => 3,
+                'prices_by_zone_id' => [
+                    1 => 3,
+                    4 => 5,
+                ],
+            ],
+            [
+                'from' => 2,
+                'to' => 5,
+                'prices_by_zone_id' => [
+                    1 => 2,
+                    4 => 5,
+                ],
+            ],
+        ]];
+    }
+
+    private function getValidDataForCommandCreation()
+    {
+        return [
+            'localized_names' => [1 => 'My carrier'],
+            'localized_delays' => [1 => 'pickup in store'],
+            'speed_grade' => 0,
+            'tracking_url' => 'http://example.com/track.php?num=@',
+            'shipping_cost_included' => true,
+            'shipping_method' => ShippingMethod::SHIPPING_METHOD_PRICE,
+            'tax_rules_group' => 1,
+            'out_of_range_behavior' => OutOfRangeBehavior::APPLY_HIGHEST_RANGE,
+            'shipping_ranges' => [
+                [
+                    'from' => 1,
+                    'to' => 2,
+                    'prices_by_zone_id' => [
+                        3 => 1,
+                        4 => 2,
+                    ],
+                ],
+                [
+                    'from' => 2,
+                    'to' => 3,
+                    'prices_by_zone_id' => [
+                        3 => 2,
+                        5 => 1,
+                    ],
+                ],
+            ],
+            'width' => 15,
+            'height' => 10,
+            'depth' => 20,
+            'weight' => 20.1,
+            'associated_group_ids' => [1, 2],
+            'associated_shop_ids' => [1],
+        ];
+    }
+
+    private function createCommandFromArray(array $data, bool $freeShipping)
+    {
+        $command = AddCarrierCommand::withPricedShipping(
```

Comment:
IMO it would be more efficient to put this into `else` condition instead of overwriting it later on